src/base/preferences.cpp

```diff
@@ -281,6 +281,16 @@ void Preferences::setPreventFromSuspendWhenSeeding(const bool b)
     setValue("Preferences/General/PreventFromSuspendWhenSeeding", b);
 }
 
+void Preferences::setTheme(UIThemeManager::Theme theme)
+{
+    setValue("Preferences/General/Theme", static_cast<int>(theme));
```

Comment:
you are casting `theme` to `int` eventually, you should just make the parameter `int` type.

Reply:
then what's the point of using `enum class`

Reply:
Ha, I should be the one asking that problem. Joking aside, this specific function is feeding to/from `m_ui->comboTheme`. The combobox only understands `int` and thus it doesn't need to know (or use) your customized enum class, isn't?
Plugging enum into the middle of it feels unnatural for me.

Reply:
```
 enum Theme
 {
         Default,
         Custom
 };

 ...

 m_ui->comboTheme->setCurrentIndex(Preferences::instance()->theme());
```

isn't above looks more cryptic?

Reply:
If I were to do it:
```c++
m_ui->comboTheme->setCurrentIndex((Preferences::instance()->getGUIThemeSelection() == 0) ? 0 : 1);
```
Also you introduced a *potential* undefined behavior by `static_cast<UIThemeManager::Theme>` in `Preferences::theme()` which occurs when the value is large to an extent. That's why I prefer making getters/setters as simple as possible and move the logic of verifying the value validness outside of it instead.

>but the latter is not intituitive.

defying your own idea...!?

Reply:
Then I have to write some where what 0 and 1 means in this context.
IMHO I like "current" way, but will change if you insist. You're the boss XD

Reply:
Oh ho that what `selection` implies here.
Ok now I understood.

Reply:
Or perhaps in the current context you can work around by using `bool`:
`void Preferences::setGUIUseCustomTheme(bool)`. It is either custom or not (the default).

Reply:
I personally like named constants in my code. This way clearly shows the intent.
Having an enum cleanly defines different types of `themes` available. 
IMHO I find you're both suggested ways more complicated. I myself took some time to understand
what you meant by `selection` but maybe I'm little dumb.
idk maybe I'm wrong. @glassez?